src/routes/Profile/Competences/Competences.tsx

```diff
@@ -212,27 +295,43 @@ const Competences = () => {
         simul accusata no ius. Volumus corpora per te, pri lucilius salutatus iracundia ut. Mutat posse voluptua quo cu,
         in albucius nominavi principes eum, quem facilisi cotidieque mel no.
       </p>
-      {groupBy === GROUP_BY_SOURCE && (
-        <GroupBySource
-          filters={selectedFilters}
-          filterKeys={filterKeys}
-          locale={locale}
-          osaamiset={osaamiset}
-          deleteOsaaminen={deleteOsaaminen}
-          isOsaaminenVisible={isOsaaminenVisible}
-        />
-      )}
-      {groupBy === GROUP_BY_THEME && <></>}
-      {groupBy === GROUP_BY_ALPHABET && (
-        <GroupByAlphabet
-          filters={selectedFilters}
-          filterKeys={filterKeys}
-          locale={locale}
-          osaamiset={osaamiset}
-          deleteOsaaminen={deleteOsaaminen}
-          isOsaaminenVisible={isOsaaminenVisible}
-        />
-      )}
+      <div>
+        {!sm && (
+          <Modal
+            open={showFilters}
+            onClose={() => setShowFilters(false)}
+            content={<Filters />}
+            footer={
+              <div className="flex flex-row justify-end gap-4">
+                <Button variant="white" label="Sulje" onClick={() => setShowFilters(false)} />
+              </div>
+            }
+          />
+        )}
+        {groupBy === GROUP_BY_SOURCE && (
+          <GroupBySource
+            filters={selectedFilters}
+            filterKeys={filterKeys}
+            locale={locale}
+            osaamiset={osaamiset}
+            deleteOsaaminen={deleteOsaaminen}
+            isOsaaminenVisible={isOsaaminenVisible}
+            mobileFilterOpenerComponent={<MobileFilterButton />}
```

Comment:
I think it would be cleaner to just have the MobileFilterButton in a separate file instead of passing it around